consensus/src/pending_votes.rs

```diff
@@ -494,39 +526,167 @@ mod tests {
             &signers[1],
         )
         .unwrap();
+        vote_data_2_author_1.set_verified();
         assert_eq!(
-            pending_votes.insert_vote(&vote_data_2_author_1, &validator),
+            pending_votes.insert_vote(&vote_data_2_author_1, &validator_verifier),
             VoteReceptionResult::VoteAdded(1)
         );
 
         // two votes for the ledger info -> NewQuorumCertificate
         let vote_data_2_author_2 =
             Vote::new(vote_data_2, signers[2].author(), li2, &signers[2]).unwrap();
-        match pending_votes.insert_vote(&vote_data_2_author_2, &validator) {
+        vote_data_2_author_2.set_verified();
+        match pending_votes.insert_vote(&vote_data_2_author_2, &validator_verifier) {
+            VoteReceptionResult::NewQuorumCertificate(qc) => {
+                assert!(qc
+                    .ledger_info()
+                    .check_voting_power(&validator_verifier)
+                    .is_ok());
+            },
+            _ => {
+                panic!("No QC formed.");
+            },
+        };
+    }
+
+    #[test]
+    fn test_qc_aggregation_with_unverified_votes() {
+        ::aptos_logger::Logger::init_for_testing();
+
+        // set up 4 validators
+        let (signers, validator_verifier) = random_validator_verifier(7, Some(3), false);
+        let mut pending_votes = PendingVotes::new();
+
+        // create random vote from validator[0]
+        let mut li = random_ledger_info();
+        let vote_data = random_vote_data();
+        li.set_consensus_data_hash(vote_data.hash());
+
+        let mut partial_sigs = PartialSignatures::empty();
+
+        let vote_0 = Vote::new(
+            vote_data.clone(),
+            signers[0].author(),
+            li.clone(),
+            &signers[0],
+        )
+        .unwrap();
+
+        let vote_1 = Vote::new(
+            vote_data.clone(),
+            signers[1].author(),
+            li.clone(),
+            &signers[1],
+        )
+        .unwrap();
+
+        let vote_2 = Vote::new_with_signature(
+            vote_data.clone(),
+            signers[2].author(),
+            li.clone(),
+            bls12381::Signature::dummy_signature(),
+        );
+
+        let vote_3 = Vote::new(
+            vote_data.clone(),
+            signers[3].author(),
+            li.clone(),
+            &signers[3],
+        )
+        .unwrap();
+
+        let vote_4 = Vote::new(
+            vote_data.clone(),
+            signers[4].author(),
+            li.clone(),
+            &signers[4],
+        )
+        .unwrap();
+
+        // first time a new vote is added -> VoteAdded
+        assert_eq!(
+            pending_votes.insert_vote(&vote_0, &validator_verifier),
+            VoteReceptionResult::VoteAdded(1)
+        );
+        partial_sigs.add_signature(signers[0].author(), vote_0.signature().clone());
+
+        // same author voting for the same thing -> DuplicateVote
+        vote_0.set_verified();
```

Comment:
same q here

Reply:
Just wanted to make sure that the data structure works when it receives a mix of verified and unverified signatures.